plexapi/video.py

```diff
@@ -925,15 +922,38 @@ def _loadData(self, data):
 
         # If seasons are hidden, parentKey and parentRatingKey are missing from the XML response.
         # https://forums.plex.tv/t/parentratingkey-not-in-episode-xml-when-seasons-are-hidden/300553
-        if self.skipParent and data.attrib.get('parentRatingKey') is None:
-            # Parse the parentRatingKey from the parentThumb
-            if self.parentThumb and self.parentThumb.startswith('/library/metadata/'):
-                self.parentRatingKey = utils.cast(int, self.parentThumb.split('/')[3])
-            # Get the parentRatingKey from the season's ratingKey
-            if not self.parentRatingKey and self.grandparentRatingKey:
-                self.parentRatingKey = self.show().season(season=self.parentIndex).ratingKey
-            if self.parentRatingKey:
-                self.parentKey = f'/library/metadata/{self.parentRatingKey}'
+        # Use cached properties below to return the correct values if they are missing to avoid auto-reloading.
+        self._parentKey = data.attrib.get('parentKey')
+        self._parentRatingKey = utils.cast(int, data.attrib.get('parentRatingKey'))
+        self._parentThumb = data.attrib.get('parentThumb')
+
+    @cached_property
+    def parentKey(self):
+        """ Returns the parentKey. Refer to the Episode attributes. """
+        return self._parentKey or f'/library/metadata/{self.parentRatingKey}'
+
+    @cached_property
+    def parentRatingKey(self):
+        """ Returns the parentRatingKey. Refer to the Episode attributes. """
+        if self._parentRatingKey is not None:
+            return self._parentRatingKey
+        # Parse the parentRatingKey from the parentThumb
+        if self._parentThumb and self._parentThumb.startswith('/library/metadata/'):
+            return utils.cast(int, self._parentThumb.split('/')[3])
+        # Get the parentRatingKey from the season's ratingKey
+        return self._season.ratingKey
+
+    @cached_property
+    def parentThumb(self):
+        """ Returns the parentThumb. Refer to the Episode attributes. """
+        return self._parentThumb or self._season.thumb
+
+    @cached_property
+    def _season(self):
+        """ Returns the :class:`~plexapi.video.Season` object by querying for the show's children. """
+        return self.fetchItem(
+            f'{self.grandparentKey}/children?excludeAllLeaves=1&index={self.parentIndex}'
```

Comment:
It seems this request returns _500 internal error_ when `index=None`
See https://github.com/Taxel/PlexTraktSync/issues/1631